cli/spec.py

```diff
@@ -0,0 +1,260 @@
+                conflict_resolver=utils.ConflictResolver.none_resolver)
+
+    # replace defaults with cli
+    utils.dict_merge(res_args, defaults,
+                     conflict_resolver=utils.ConflictResolver.none_resolver)
+
+    # check if we have all the required arguments defined at the end of merge
+    unset_args = []
+    only_args = []
+    for arg_name, arg_param in command_args.iteritems():
+        if 'required' in arg_param and \
+                arg_param['required'] and \
+                res_args.get(arg_name, None) is None:
+            unset_args.append(arg_name)
+        if 'requires_only' in arg_param:
+            only_args.extend(arg_param['requires_only'])
+    if len(only_args) > 0:
+        intersection = set(unset_args).intersection(set(only_args))
+        if len(intersection) > 0:
+            raise exceptions.IRConfigurationException(
+                "Required only input arguments {} are not set!".format(
+                    intersection))
+    elif len(unset_args) > 0:
+        raise exceptions.IRConfigurationException(
+            "Required input arguments {} are not set!".format(
+                unset_args))
+    return res_args
+
+
+def _get_command_line_args(args, config, module_name):
+    """
+    Gets the arguments values from the command line.
+    """
+    clg_options, global_opts, subcommand_opts = \
+        _load_spec_as_dict(module_name, config)
+    cmd = clg.CommandLine(clg_options)
+    res_args = vars(cmd.parse(args))
+    return res_args, subcommand_opts
+
+
+def _load_spec_as_dict(module_name, config):
+    """
+    Loads the spec file as dict.
+
+    Returns tuple:
+        dict of options and groups for clg
+        dict of options without subcommands
+        dict of options for subcommands
+    """
+    clg_options = _get_specs(module_name, config)
+    global_options, subcommands_options = _process_options(clg_options)
+    return clg_options, global_options, subcommands_options
+
+
+def _process_options(spec_dict, lookup=False, trim=False):
+    """
+    Goes through all the spec options modifies them by removing
+     some options parameters (like defaults)
+
+    :param spec_dict: the dictionary with key obtained from spec file
+    :param lookup: specifies whether the options lookup is required.
+    :param trim: specifies whether the certain option
+        parameters should be removed
+    """
+    global_options = {}
+    subcommand_options = {}
+    _process_flat_options(global_options, spec_dict, lookup, trim)
+    _process_groups_options(global_options, spec_dict, lookup, trim)
+    # collect sub parser global_options
+    if 'subparsers' in spec_dict:
+        for subparser_name, params in spec_dict['subparsers'].iteritems():
+            subcommand_options[subparser_name], _ = _process_options(
+                params, lookup=True, trim=True)
+
+    return global_options, subcommand_options
+
+
+def _process_groups_options(options, spec_dict, lookup, trim):
```

Comment:
Please consider more meaningful name (ex: 'get_parser_group_options')
Also, create and return dict object inside this function instead of receiving it from the outside.